README.md

```diff
@@ -15,3 +15,38 @@ Run `cmake` to configure the build environment and then `make` to build
 To install, use `make install`
 
     sudo make install
+
+## Writing contract files
+Elementary OS's Pantheon's file manager supports adding options to the context menu by the way of Contract files.
+
+Contract files can be made system wide available by adding them to:
+`/usr/share/contractor`
+or for specific users by adding them to:
+`~/.local/share/contractor`
+
+Simple .contract file example:
+```
+[Contractor Entry]
+Name=Mount
+MimeType=application/x-cd-image;application/x-raw-disk-image
+Exec=gnome-disk-image-mounter %f
+```
+
+- `Name`: Text displayed in the right click context menu.
+- `MimeType`: Mimetype(s) of files it should be shown for.
+- `Exec`: Command to execute. [More info](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables).
+
+Tips:
+- Use `pkexec` to ask for root permissions. Example: `Exec=pkexec chmod +x %U`
+- To execute more complex scripts, put the script in a .sh file and call it.
+
+Examples:
```

Comment:
I think it would be better to link to examples in the source trees of actual apps than to StackExchange posts which probably won't be maintained as well

Reply:
Like contracts using Catfish (for search), Thunar (file renaming) or gnome-disk-image-mounter (mounting images)? But pointing just to those projects only get's you halfway there, you'd still need to figure out how to turn those into contract files, which arguments to use etc. I'm not sure what would be the best way to distribute these. A separate github repo? That would open it up for issues and pull requests. I can't imagine them ending up in the app's repositories themselves, since they seem to be elementary specific?

Reply:
You could use apps that do supply contract files like https://github.com/elementary/switchboard-plug-pantheon-shell/blob/master/set-wallpaper-contract/set-wallpaper.contract.in or https://github.com/elementary/mail/blob/master/data/mail-attach.contract

Reply:
That's great, but I'm not going to create 5 apps to share examples. I'm not even sure what the AppCenter's policy would be or should be on a contract only app. I understand this is a more complicated subject, I'll remove the examples from this pull request and open a new one to discuss a good way to share contract file examples.